web/src/test/java/org/hillview/HistogramAccuracyTest.java

```diff
@@ -160,14 +175,71 @@ private double computeAccuracy(PrivateHistogram ph, int totalLeaves) {
         return abstot / (double) n;
     }
 
-    private double computeSingleColumnAccuracy(String col, DoubleColumnQuantization dq, double epsilon, IDataSet<ITable> table,
+    /**
+     * Compute the absolute and L2 error vs. ground truth for an instantiation of a private heat map
+     * averaged over all possible range queries (every rectangle).
+     *
+     * @param ph The heat map whose accuracy we want to compute.
+     * @param totalXLeaves The "global" number of leaves (in the x-dimension) in case this heat map is computed only on a zoomed-in range.
+     *                    This is needed to correctly compute the amount of noise to add to each leaf.
+     * @param totalYLeaves same as totalXLeaves but for y-axis
+     * @return The average per-query absolute error.
+     */
+    private Double computeAccuracy(PrivateHeatmap ph, int totalXLeaves, int totalYLeaves) {
```

Comment:
I expect that you will want to use this kind of code in the benchmarking program, so it should probably become a method - would it work in the PrivateHeatmap?

We should also add the code that is used for measurements to the DPBenchmarks class - perhaps with more command-line arguments to choose one of the benchmarks to run. Ideally we should be able to run all the measurements with just one program.

Reply:
Hmm, I don't think this should be functionality of the histogram/heatmap as it's really benchmarking code, e.g. we'll never expose this benchmark information to a user.

I'll move this to a benchmark class.